job-server/src/main/scala/spark/jobserver/WebApi.scala

```diff
@@ -497,6 +497,9 @@ class WebApi(system: ActorSystem,
                           resultToByteIterator(jobReport, s.toIterator))
                       case _ => ctx.complete(jobReport ++ resultToTable(result))
                     }
+                  case NoSuchJobId =>
+                    ctx.complete(500, errMap("Status, but no result with job ID "
```

Comment:
I think 500 wouldn't be the best error code here since there's no error.  In any case 503 which means a temporary error might be better. But I would rather reply a 200 with a text specifying that they should retry